daemon/config.go

```diff
@@ -75,6 +75,8 @@ func (config *Config) InstallFlags() {
 	config.Ulimits = make(map[string]*ulimit.Ulimit)
 	opts.UlimitMapVar(config.Ulimits, []string{"-default-ulimit"}, "Set default ulimits for containers")
 	flag.StringVar(&config.LogConfig.Type, []string{"-log-driver"}, "json-file", "Default driver for container logs")
+	config.LogConfig.Config = make(map[string]string)
```

Comment:
In that case, I'd have to pass a pointer. Which was my original implementation of this. But, I changed it to this approach after comments from @LK4D4